PhaseAndSave3DLineNarrowV1.ipf

```diff
@@ -583,16 +583,19 @@ Function Open_File_source(ctrlName): ButtonControl
 		if ((Abs(source_wave_real[ipntnum])<1e-9)&&(Abs(source_wave_imag[ipntnum])<1e-9))
 			InZeroBool=1
 			iAtEndOfFirstZeroBurst=ipntnum //use this to hold the current Zero value
+			break
 		endif
-		//debugging...worked...no longer needed//Print "ipntnum=", ipntnum,  "Re=", Abs(source_wave_real[ipntnum]), "Im=", Abs(source_wave_imag[ipntnum]), "InZeroBool=", InZeroBool
 		ipntnum+=1
-	While ((((Abs(source_wave_real[ipntnum])<1e-9)&&(Abs(source_wave_imag[ipntnum])<1e-9)) || (InZeroBool==0)) && (ipntnum<source_numpnts))
+	While (ipntnum<source_numpnts)
 
 	if (ipntnum == source_numpnts)
 		iAtEndOfFirstZeroBurst = ipntnum
 	endif
 
-	Variable/G Acq=(iAtEndOfFirstZeroBurst+1), PntToSetPh0=(Acq/2-1)
+	Variable/G Acq=iAtEndOfFirstZeroBurst, PntToSetPh0=floor(Acq/2)
+	if(Acq == source_numpnts)
+		PntToSetPh0 = 1 // if only 1 acq block, put PntToSetPh0 at the second point by default
+	endif
```

Comment:
I don't know what use case the "iAtEndOfFirstZeroBurst+1" was intended for, but it's never really worked for my data. So I got rid of the +1 and added a floor to PntToSetPh0, so no more fractional indices.

Also, in the case that the while loop finished without breaking, we don't actually have piecewise acquired data, so it makes more sense to set PntToSetPh0 to the beginning of the wave.